build: update the recommended `Dockerfile` for VSCode remote development

gkalpak opened the pull request:
This PR updates the recommended `Dockerfile` for VSCode remote development to take into account recent changes in our dev workflow (updated Node.js version, vendored yarn, modified CircleCI config).

It also adds a note in the CircleCI config to keep the recommended `Dockerfile` up-to-date with the Node.js version used on CI.

gkalpak commented:
Blocked on #41374.

destus90 commented on .circleci/config.yml:
```diff
@@ -65,6 +65,7 @@ var_10: &only_on_master
 # **NOTE 1**: Pin to exact images using an ID (SHA). See https://circleci.com/docs/2.0/circleci-images/#using-a-docker-image-id-to-pin-an-image-to-a-fixed-version.
 #             (Using the tag in not necessary when pinning by ID, but include it anyway for documentation purposes.)
 # **NOTE 2**: If you change the version of the docker images, also change the `cache_key` suffix.
+# **NOTE 3**: If you change the version of Node.js provided by the dcocker images, also update `.devcontainer/recommended-Dockerfile` to match the new version.
```
```suggestion
# **NOTE 3**: If you change the version of Node.js provided by the docker images, also update `.devcontainer/recommended-Dockerfile` to match the new version.
```

gkalpak replied:
Ooops 😇 Thx, @destus90!

gkalpak committed:
build: update the recommended `Dockerfile` for VSCode remote development

This commit updates the recommended `Dockerfile` for VSCode remote
development to take into account recent changes in our dev workflow
(updated Node.js version, vendored yarn, modified CircleCI config).

It also adds a note in the CircleCI config to keep the recommended
`Dockerfile` up-to-date with the Node.js version used on CI.

josephperrott reviewed (approved):
LGTM
